### ISSUE TITLE
Adjacency matrix, DiGraph and self-loops

### ISSUE THREAD

**Issue by gmarcais** [JuliaGraphs/LightGraphs.jl#595]
For a node which has a self-loop, the diagonal element of the adjacency matrix is set to 2. This convention is fine for undirected graphs. On the other hand, for directed graphs, I think the value on the diagonal should be 1.
In particular, setting the diagonal to 2 for a directed graph breaks the property that the row-sum is the out-degree of a node, and the column-sum is the in-degree.

```Julia
julia> g = DiGraph(1); add_edge!(g, 1, 1)
true
julia> [sum(adjacency_matrix(g), 1) sum(adjacency_matrix(g), 2)]
1×2 Array{Int64,2}:
 2  2
julia> [indegree(g, 1) outdegree(g, 1)]
1×2 Array{Int64,2}:
 1  1
```

**Comment by sbromberger** [comment]
This should be an easy PR if you'd care to give it a shot (should be a one-line change). The relevant lines are in `linalg/spectral` around line 55.

**Comment by gmarcais** [comment]
OK, I did create PR #596. Hopefully done correctly. I wrote it agains v0.7.4 as it is the newest version that worked with Julia 0.5.1.

**Comment by sbromberger** [comment]
Closed via #596.
